src/Build/Evaluation/ProjectRootElementCache.cs

```diff
@@ -273,7 +273,7 @@ private bool IsInvalidEntry(string projectFile, ProjectRootElement projectRootEl
             try
             {
                 // We are about to load. Take a per-file lock to prevent multiple threads from duplicating the work multiple times.
```

Comment:
Alternatively, you could have changed the comment from "Take a per-file lock" to "Take a global lock" :smile: